src/coreclr/jit/lsra.cpp

```diff
@@ -2718,12 +2717,6 @@ bool LinearScan::isMatchingConstant(RegRecord* physRegRecord, RefPosition* refPo
             }
             break;
         }
-
-        case GT_CNS_VEC:
-        {
-            return GenTreeVecCon::Equals(refPosition->treeNode->AsVecCon(), otherTreeNode->AsVecCon());
-        }
```

Comment:
Instead of reverting, would it be feasible for this to be the following?
```cpp
return (otherTreeNode->gtType != TYP_SIMD32) && GenTreeVecCon::Equals(refPosition->treeNode->AsVecCon(), otherTreeNode->AsVecCon());
```

Asking as the original change had some measured perf wins and since it looks like the bug only impacts `TYP_SIMD32` due to the need to save/restore the upper halves (meaning I think it shouldn't impact `TYP_SIMD8/12/16`).

Reply:
If we want to scope it, we only need to do it for platforms where `FEATURE_PARTIAL_SIMD_CALLEE_SAVE` is defined (win-amd64, arm64, LoongArch64). Presumably there is no issue on x86?

`varTypeNeedsPartialCalleeSave` would need to be used.

Reply:
However, I still think it's probably "too late" to ensure correctness, on all platforms.

Reply:
`varTypeNeedsPartialCalleeSave ` check seems reasonable to include, but I agree with @BruceForstall that the simplest thing would be to revert the change at this point.

Reply:
👍, just thought I'd throw it out as a suggestion given this will have negative impact on Arm64.

I expect we'll see this one https://github.com/dotnet/perf-autofiling-issues/issues/5976 and a few of the other code paths that use vector constants regress a bit.

Reply:
I looked at the spmi-diffs and there are few in benchmarks. I tried the change to just use `varTypeNeedsPartialCalleeSave ` and didn't see as many spmi-diffs in benchmarks, asp locally. I will let CI run and check other diffs.

Reply:
There is no change in spmi diff on Arm64 if we just disable `isMatchingConstant` for `varTypeNeedsPartialCalleeSave`. On x64, there is slight difference (earlier in benchmark collection, 12 methods regressed and now 5). 

> it looks like the bug only impacts TYP_SIMD32 due to the need to save/restore the upper halves (meaning I think it shouldn't impact TYP_SIMD8/12/16).

@tannergooding , even Arm64 needs save/restore upper half as per this comment, so it could be problematic there too, right?

https://github.com/dotnet/runtime/blob/519d1e31184bcdbbb4cbf0a5688cc80366468bd3/src/coreclr/jit/compiler.h#L7439-L7445

Reply:
> What was the regression on Arm64 if we revert the entire PR?

There is no change between total revert and limiting it to just `varTypeNeedsPartialCalleeSave` on Arm64.

https://dev.azure.com/dnceng/public/_build/results?buildId=1951028&view=ms.vss-build-web.run-extensions-tab

Reply:
> (https://github.com/ARM-software/abi-aa/blob/main/aapcs64/aapcs64.rst#simd-and-floating-point-registers):

Thanks for this incredible resource. I didn't know about it.